app.rb

```diff
@@ -0,0 +1,166 @@
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'book'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  # Show menu for users
+  def show_menu
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts "7 - Exit\n\n"
+    gets.chomp
+  end
+
+  # Method that handles user choice
+  def run
+    user_choice = show_menu
+    case user_choice
+    when '1'
+      action_list_books
+    when '2'
+      action_list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rental
+    else
+      puts 'Thanks for using this app ...'
+      exit
+    end
+  end
+
+  # List all books
+  def list_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  # Action list all Books
+  def action_list_books
+    list_books
+    puts 'Press enter to continue ...'
+    gets.chomp
+    run
+  end
+
+  # Permission access for Person (Teacher or Student)
+  def permission(my_char)
+    case my_char
+    when 'N' || 'n'
+      false
+    when 'Y' || 'y'
+      true
+    end
+  end
+
+  # List all People
+  def list_people
+    @people.each_with_index do |person, index|
+      # puts "#{index}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
```

Comment:
- [ ] Kindly remove commented out code for a clean code base. Besides, you can always refer to your previous code by checking out any of your commits.